### PR TITLE
Update root NS record on zone creation rather than attempt to create it

### PR DESCRIPTION
**Pull request by jmeyers35**
Same idea as https://github.com/octodns/octodns-ultra/pull/28: we ran into https://github.com/octodns/octodns-ns1/issues/48 and fixed it with these changes.

### CODE REVIEW

File: octodns_ns1/__init__.py [hunk 1]
```diff
@@ -1760,9 +1778,12 @@ def _apply_Update(self, ns1_zone, change):
...
         self._client.records_update(zone, domain, _type, **params)
-        # If we're cleaning up we need to send in the old record since it'd
-        # have anything that needs cleaning up
-        self._monitors_gc(change.existing, active_monitor_ids)
+        # It's possible change.existing is None because in the case of zone creation, we swap out the NS record Create for an Update, but we don't set the existing
+        # record (see _force_root_ns_update).
+        if change.existing is not None:
+            # If we're cleaning up we need to send in the old record since it'd
+            # have anything that needs cleaning up
+            self._monitors_gc(change.existing, active_monitor_ids)
```
Comment (jmeyers35) [inline octodns_ns1/__init__.py:1786]: This is the bit I'm most unsure about - would like some feedback on if this is okay to do, or if not what the recommended approach to populating `change.existing` when we hit this case would be.

Comment (ross) [reply octodns_ns1/__init__.py:1786]: Assuming the Update instantiation happens I think it'd make sense to drive the logic off of change.existing as you have here.

File: octodns_ns1/__init__.py [hunk 2]
```diff
@@ -1744,6 +1744,24 @@ def _extra_changes(self, desired, changes, **kwargs):
...
+            if (
+                change.record.name == ''
+                and change.record._type == 'NS'
+                and isinstance(change, Create)
+            ):
+                self.log.info(
+                    '_force_root_ns_update: found root NS record creation, changing to update'
+                )
+                changes[i] = Update(None, change.record)
+        return
```
Comment (ross) [inline octodns_ns1/__init__.py:1763]:
```suggestion
        return changes
```

Minor tweak to just make it inherently obvious that _force_root_ns_update is going to be modifying changes.

File: octodns_ns1/__init__.py [hunk 3]
```diff
@@ -1800,6 +1821,7 @@ def _apply(self, plan):
                 raise
             self.log.debug('_apply:   no matching zone, creating')
             ns1_zone = self._client.zones_create(domain_name)
+            self._force_root_ns_update(changes)
```
Comment (ross) [inline octodns_ns1/__init__.py:1824]:
```suggestion
            changes = self._force_root_ns_update(changes)
```

**Review by ross — APPROVED** [review]

**Comment by ross** [comment]
Thanks!

/cc Fixes https://github.com/octodns/octodns-ns1/issues/48

**Pull request merged**